scripts/parallel_pono.py

```diff
@@ -167,21 +175,20 @@ def main() -> int:
             if process.poll() is not None:  # process has finished
                 if args.summarize:
                     runtime = end_time - start_times[name]
-                    cmd = cast(list[str], process.args)
+                    cmd = cast("list[str]", process.args)
                     summarize(args.summarize, name, process.returncode, runtime, cmd)
-                if process.returncode in SOLVED_RETURN_CODES:
-                    if process.stdout is None:
-                        logger.warning(f"{name} has no stdout")
-                        print(ReturnCode(process.returncode).name.lower())
-                    else:
-                        print(process.stdout.read())
-                    if args.witness_file:
-                        witnesses[name].rename(args.witness_file)
-                    return process.returncode
-                else:
+                if process.returncode not in SOLVED_RETURN_CODES:
                     del processes[name]
-                    clean_up({name: process}, witnesses, args.verbose)
+                    clean_up({name: process}, witnesses, verbose=args.verbose)
                     break
+                if process.stdout is None:
+                    logger.warning("%s has no stdout", name)
+                    print(ReturnCode(process.returncode).name.lower())
+                else:
+                    print(process.stdout.read())
+                if args.witness_file:
+                    witnesses[name].rename(args.witness_file)
```

Comment:
IIRC, we wanted to use `shutil.move` instead of `Path.rename` to ensure it works across filesystems.